bokeh/server/tornado.py

```diff
@@ -23,6 +24,18 @@
 from .connection import ServerConnection
 from .application_context import ApplicationContext
 
+def _whitelist(handler_class):
+    if hasattr(handler_class.prepare, 'patched'):
+        return
+    old_prepare = handler_class.prepare
+    def _prepare(self, *args, **kw):
+        if self.request.host not in self.application._hosts:
+            raise HTTPError(403)
+        old_prepare(self, *args, **kw)
```

Comment:
it's probably good to `return old_prepare` so we get any return value it may have now or in the future. If it returns None, no harm done.